src/encord_active/lib/common/utils.py

```diff
@@ -54,6 +54,7 @@ class ProjectMeta(TypedDict):
     project_title: str
     ssh_key_path: str
     has_remote: bool
+    metrics: dict[str, str]
```

Comment:
can this keep the actual metric instead of the module name?

Reply:
I don't understand well this question.

Reply:
I assume that the type here is `dict[str, str]` which maps from metric name to module? e.g. `"Blur": "src/encord_active/lib/metrics/heuristic/img_features.py"`. 

So what I'm suggesting is to change it to `dict[str, Metric]` or even `Set[Metric]`

If I'm wrong please correct me and disregard.

Reply:
From my pov, we can't store metrics instances per se because they are not persistent. But we can try and add module copies inside a folder in the project and open those instead of the "remote" ones (like what pip does). Also, next step would be upgrade (update) a metric and after that add metric versioning. I would like to allow at the first/second instance editable metrics so you don't have to update them every time you make a change (like pip install -e path/to/package). Main point is that we don't want to lose track from where the metric comes from, either to update it or to have it as editable (making it easier to download metrics from url afterwards).

On another note, highly likely metric metadata will migrate to be created at add time instead of run.

Reply:
atm, if we use of `Metric`/`SimpleMetric` in `fetch_project_meta` level (where any transformation from the I/O version to the actual instance should happen) then a very tangled cyclic dependency is created `I/O` -> `Metric` -> `Iterator` -> `Project` -> `fetch_project_meta` -> `I/O` & `Metric`. I tried to remove it in vain last Friday.
Btw, from my pov that cyclic dependency is the reason why the code of project's metadata is already scattered almost everywhere.

Reply:
> From my pov, we can't store metrics instances per se because they are not persistent. But we can try and add module copies inside a folder in the project and open those instead of the "remote" ones (like what pip does). Also, next step would be upgrade (update) a metric and after that add metric versioning. I would like to allow at the first/second instance editable metrics so you don't have to update them every time you make a change (like pip install -e path/to/package). Main point is that we don't want to lose track from where the metric comes from, either to update it or to have it as editable (making it easier to download metrics from url afterwards).

I'm not sure I'm on board with this idea. I don't think we should be implementing metric version management into the the project, instead, I would suggest providing the metrics as an actual dependency and publishing them as a separate package (which we anyways want to do for the sake of modularity)

Reply:
> atm, if we use of `Metric`/`SimpleMetric` in `fetch_project_meta` level (where any transformation from the I/O version to the actual instance should happen) then a very tangled cyclic dependency is created `I/O` -> `Metric` -> `Iterator` -> `Project` -> `fetch_project_meta` -> `I/O` & `Metric`. I tried to remove it in vain last Friday. Btw, from my pov that cyclic dependency is the reason why the code of project's metadata is already scattered almost everywhere.

I understand that problem, we should solve it, but not in the scope of this PR probably.

Reply:
I moved the metrics metadata from the project_meta file to its own file so we can discuss its structure there.